src/Microsoft.AspNetCore.ResponseCaching/Internal/IResponseCacheKeyProvider.cs

```diff
@@ -3,9 +3,9 @@
 
 using System.Collections.Generic;
 
-namespace Microsoft.AspNetCore.ResponseCaching
+namespace Microsoft.AspNetCore.ResponseCaching.Internal
 {
-    public interface IResponseCacheKeyProvider
+    internal interface IResponseCacheKeyProvider
```

Comment:
Sort of. I don't think there are a specific set of strict rules for what goes into `*.Internal` namespace. I interpreted it as: these types are internal and users shouldn't need to access them. Some of the types are enforced this way by having the internal accessibility modifier. Other types are still public so that if users really need them, they can use them. Leaving it in the internal namespace adds a big asterisk though, indicating that these types should be used with caution and may change between minor version updates.